### DISCUSSION TITLE
Broken QCoDeS install from conda (was Problem running Qcodes for kEYSIGHT B1505A)

### DISCUSSION THREAD

**Discussion by Dianebonk** [microsoft/Qcodes#5926] [Q&A] [answered]
Hello, 

 I run Keysight connection expert and connect my instrument. I can send the basic command and get answer for the qcodes example for B1505A for example with pyvisa : 

`import pyvisa

# Initialize the PyVISA resource manager
rm = pyvisa.ResourceManager()

# List available resources (including GPIB instruments)
print(rm.list_resources())

# Open a connection to your GPIB instrument
inst = rm.open_resource('GPIB0::17::INSTR')  # Adjust the address as per your instrument

# Query the instrument for identification

print(inst.query("*IDN?"))
print(inst.query("UNT?"))
inst_query_delay=0.1`

But, I'm encountering issues with driver missing for B1505 and also the file dataset when I run the code I have found in this website with Qcode: [https://microsoft.github.io/Qcodes/examples/driver_examples/Qcodes%20example%20with%20Keysight%20B1500%20Parameter%20Analyzer.html] 

`from IPython.display import Markdown, display
from matplotlib import pyplot as plt
from pyvisa.errors import VisaIOError

import qcodes as qc
from qcodes.dataset import (
    Measurement,
    initialise_database,
    load_or_create_experiment,
    plot_dataset,
)
from qcodes.instrument_drivers.Keysight import KeysightB1500
from qcodes.instrument_drivers.Keysight.keysightb1500 import MessageBuilder, constants`

I can't import qcode.dataset neither  import KeysightB1500. An I am blocked in these line and can't run the rest of the code as I can't import the librairies. I am using Spyder with Ananconda and I have installed qcode in ananconda prompt using : conda install qcodes.

Does anyone have this problem once? How can I solve it? 

Thanks

**Comment by Dianebonk** [comment]
For more info I get the error: 

No module named 'qcodes.dataset

**Comment by jenshnielsen** [comment]
@Dianebonk From your description it sounds like your problem is a broken install of qcodes from conda. 
I took the liberty of updating your title to reflect this.

I have just created a fresh conda environment fresh and installed qcodes from conda forge and the import seems to work as expected. 

> I am using Spyder with Ananconda and I have installed qcode in ananconda prompt using : conda install qcodes.

Which conda channel are you installing qcodes from? QCoDeS is only available in conda-forge as far as I know so can you confirm that this is where you installed it from.

I would suggest that you try to create a new conda environment from fresh and try again. 

Personally I find the anaconda navigator more confusing than useful so I would

Open a (anaconda) terminal and run
```
conda create -n qcodestest python=3.11 (follow the instructions and answer yes as required)
conda activate qcodestest
conda install qcodes

```

**Comment by Dianebonk** [comment]
Thanks for your answer, I am trying to do the qcodes install, after doing the first create and activate qcodetest. 

I get error : 
![image](https://github.com/microsoft/Qcodes/assets/98905793/a869ac30-5314-4a26-95ae-ceaeac1e5b3d)



(base) PS C:\Users\BB274165> conda activate qcodestest
(qcodestest) PS C:\Users\BB274165> conda install qcodes
Collecting package metadata (current_repodata.json): failed

**CondaSSLError: OpenSSL appears to be unavailable on this machine. OpenSSL is required to
download and install packages.**

**Exception: HTTPSConnectionPool(host='repo.anaconda.com', port=443): Max retries exceeded with url: /pkgs/main/win-64/current_repodata.json (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available."))**

**Reply by jenshnielsen** [reply]
I don't know exactly what the problem here is but this looks like a fundamental issue with your conda installation. Conda fails to do basic tasks required before even installing packages. I would probably uninstall (ana)conda completely from your pc and start again. This does not look like an issue that is in any way specific to qcodes thou.

**Comment by Dianebonk** [comment]
ok thanks for replying, i'll try to reinstall my ananconda.

**Comment by Dianebonk** [comment]
I have uninstalled ananconda reinstalled it. 
Succeed installing qcodes. Channel : conda-forge

![image](https://github.com/microsoft/Qcodes/assets/98905793/b89fcd7a-1fb7-48de-b61e-59fe9d728df7)


When I run just this part of the code:
`from IPython.display import Markdown, display
from matplotlib import pyplot as plt
from pyvisa.errors import VisaIOError

import qcodes as qc
from qcodes.dataset import (
Measurement,
initialise_database,
load_or_create_experiment,
plot_dataset,
)`

 I get this error : 

![image](https://github.com/microsoft/Qcodes/assets/98905793/e5f8388d-9bc4-4601-abbc-c76b692ea511)


I don't understand why some files of qcodes is missing after my installation.

Is it possible to have the file missing and put it in the right path and run the program?

**Reply by jenshnielsen** [reply, marked as answer]
This is a very old version of qcodes  which is not compatible with the versions of the rest of the dependencies. I have no idea why you would happen but you could try installing a specific version. E.g conda install qcodes=0.44.1

**Reply by Dianebonk** [reply]
Hello @jenshnielsen, 

I finally solve my problem thanks to your recommendation.
 -Uninstall Ananconda, 
 -reinstall spyder, use miniconda as interpreter and download qcodes package using pip install in miniconda prompt, 
!!!!!!! with conda forge install I got this error : ***warning libmamba Problem type not implemented SOLVER_RULE_STRICT_REPO_PRIORITY

Now I have the last version : 
![image](https://github.com/microsoft/Qcodes/assets/98905793/fe38a31e-eccf-46ea-a5cd-65243b2f7975)


Again thank you for your help.
